### PR TITLE
Add an example of a differentiable model

### LINKED ISSUE #143: Add an example of a differentiable neuro-symbolic model
One of the main features of this library is its ability to utilize the automatic differentiation capabilities of PyTorch and thus enables hybrid neurons-symbolic models. It would be very valuable to have an example of this in the repository.

See for example: [Learning with Holographic Reduced Representations](https://proceedings.neurips.cc/paper/2021/hash/d71dd235287466052f1630f31bde7932-Abstract.html)

### PR DESCRIPTION
**Pull request by milad2073**
An Implementation of the Fully Connected model described in this [paper](https://arxiv.org/abs/2109.02157).

## Description
It is not meant to reproduce the paper's results (because the implementation of the HRR in the paper is slightly different from torchhd's HRR). It is just a good starting point, for implementing other models described in the mentioned paper and "[Multi-Label Classification With Hyperdimensional Representations](https://ieeexplore.ieee.org/document/10196438)".

Closes #143

## Checklist
- [ ] I added/updated documentation for the changes.
- [ ] I have thoroughly tested the changes.

### CODE REVIEW

File: examples/learning_with_hrr.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    
+def sparse_batch_collate(batch:list): 
+    """
+    Collate function which to transform scipy csr matrix to pytorch sparse tensor
+    """
+    data_batch, targets_batch = zip(*batch)
+
+    data_batch = vstack(data_batch).tocoo()
+    data_batch = torch.sparse_coo_tensor(data_batch.nonzero(), data_batch.data, data_batch.shape)
```
Comment (mikeheddes) [inline examples/learning_with_hrr.py:39]: I get the following warning when I run the example:
> torchhd/examples/learning_with_hrr.py:39: UserWarning: Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with numpy.array() before converting to a tensor.

**Review by mikeheddes — APPROVED** [review]
Thank you for creating this PR. This is a great addition to the examples. I made two minor comments but otherwise the code is ready to be merged.

**Comment by milad2073** [comment]
@mikeheddes I just fixed the points and pushed the changes.

**Comment by mikeheddes** [comment]
@milad2073 thank you for another great contribution to Torchhd!

**Pull request merged**